lib/storage/client.js

```diff
@@ -481,6 +481,43 @@ export default class Client extends CryptoConsumer {
     return this.internalReadNoteByName(anonAuth, noteName, encryptionKeyPair, apiUrl)
   }
 
+  async replaceNoteByName(data, recipientEncryptionKey, recipientSigningKey, options) {
+    // Automatically mix in the client id unless it is overridden in the provided options.
+    /* eslint-disable camelcase */
+    const decodedOptions = NoteOptions.decode(
+      Object.assign({ client_id: this.config.clientId }, options)
+    )
+    const encryptionKeys = new KeyPair(this.config.publicKey, this.config.privateKey)
+    let signingKeys = new KeyPair(
+      this.config.publicSigningKey,
+      this.config.privateSigningKey
+    )
+    let encryptedNote = await this.constructor.internalCreateEncryptedNote(
```

Comment:
FWIW idiomatic javascript would be to name internal functions with leading `_` vs the internal prefix

Reply:
At some point I want to refactor these internal methods out so they are hidden like the other internal helpers defined in this file...